Tests/IdentifyInterceptorTests.m

```diff
@@ -181,31 +194,43 @@ - (void)testMultipleInterceptedIdentifyIsAppliedToNextActiveIdentify {
     XCTAssertNil(event2);
     XCTAssertEqual(self->_dbHelper.getLastSequenceNumber, 2);
     XCTAssertEqual(self->_dbHelper.getInterceptedIdentifyCount, 2);
+    XCTAssertEqual(self->_dbHelper.getIdentifyCount, 0);
 
     // active identify
-    AMPIdentify *identify3 = [AMPIdentify.identify add:@"add-key" value:[NSNumber numberWithInt:1]];
-    [identify3 setOnce:@"set-once-key" value:@"set-once-value"];
+    AMPIdentify *identify3 = [AMPIdentify.identify add:@"add-key" value:@1];
+    [identify3 set:@"set-once-key" value:@"set-once-value"];
     NSMutableDictionary *event3 = [self getIdentifyEvent:identify3];
 
     event3 = [self->_identifyInterceptor intercept:event3];
 
     XCTAssertNotNil(event3);
     XCTAssertEqual(self->_dbHelper.getLastSequenceNumber, 3);
     XCTAssertEqual(self->_dbHelper.getInterceptedIdentifyCount, 0);
+    XCTAssertEqual(self->_dbHelper.getIdentifyCount, 1);
 
-    // check merged user properties
     NSMutableDictionary *userProperties3 = [AMPEventUtils getUserProperties:event3];
     NSArray *userPropertiesOperations3 = [userProperties3 allKeys];
     XCTAssertNotNil(userProperties3);
-    XCTAssertEqual(userPropertiesOperations3.count, 3);
-    BOOL hasAllOperations = [[NSSet setWithArray:userPropertiesOperations3] isEqualToSet:[NSSet setWithArray:@[AMP_OP_SET, AMP_OP_SET_ONCE, AMP_OP_ADD]]];
+    XCTAssertEqual(userPropertiesOperations3.count, 2);
+    BOOL hasAllOperations = [[NSSet setWithArray:userPropertiesOperations3] isEqualToSet:[NSSet setWithArray:@[AMP_OP_SET, AMP_OP_ADD]]];
     XCTAssertTrue(hasAllOperations);
-    XCTAssertTrue([userProperties3[AMP_OP_SET][@"set-key"] isEqualToString:@"set-value-c"]);
-    XCTAssertTrue([userProperties3[AMP_OP_SET][@"set-key-2"] isEqualToString:@"set-value-b"]);
-    XCTAssertTrue([userProperties3[AMP_OP_SET][@"set-key-3"] isEqualToString:@"set-value-d"]);
+    XCTAssertTrue([userProperties3[AMP_OP_ADD][@"add-key"] isEqualToNumber:@1]);
+    XCTAssertTrue([userProperties3[AMP_OP_SET][@"set-once-key"] isEqualToString:@"set-once-value"]);
```

Comment:
Can you also please validate that none of the set actions are applied to the "active" identify. Might be easiest to use `setOnce` on the active, then check that AMP_OP_SET is empty.

Reply:
Split the test to 2 tests.